tools/autograd/templates/VariableType.cpp

```diff
@@ -29,8 +29,14 @@
 using namespace at;
 using namespace torch::autograd::generated;
 
+${manual_functions_h}
+
 namespace torch { namespace autograd {
 
+namespace {
+${manual_functions_cpp}
```

Comment:
What is the reason why we're including ManualFunctions into VariableType? Are the forward mode formulas for e.g. `torch.abs` inlined directly into VariableType::abs ?

Reply:
Yes it is. The helper code has to be there (as we re-use values from this function) and the call to the formula itself is a one liner (either containing the formula or calling code in aten/ManualFunctions). So these are directly inlined.